src/partition/repart.c

```diff
@@ -2689,7 +2715,12 @@ static int parse_argv(int argc, char *argv[]) {
                                 arg_empty = EMPTY_REQUIRE;
                         else if (streq(optarg, "force"))
                                 arg_empty = EMPTY_FORCE;
-                        else
+                        else if (streq(optarg, "create")) {
+                                arg_empty = EMPTY_CREATE;
+                                arg_dry_run = false; /* Imply --dry-run=no if we create the loopback file
+                                                      * anew. After all we cannot really break anyone's
+                                                      * partition tables that way. */
```

Comment:
It seems very iffy to override an explicit --dry-run switch. It would be better to make `arg_dry_run` into a tristate, and only set this here where not explicitly set previously.

Reply:
indeed